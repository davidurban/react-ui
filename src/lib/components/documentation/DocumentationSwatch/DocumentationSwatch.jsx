import PropTypes from 'prop-types';
import React from 'react';
import styles from './DocumentationSwatch.scss';

const DocumentationSwatch = (props) => {
  const style = {
    backgroundColor: `var(--rui-color-${props.color})`,
  };

  return (
    <div className={styles.root} title={props.color}>
      <div className={styles.field} style={style} />
      <div className={styles.label}>
        {props.color}
      </div>
    </div>
  );
};

DocumentationSwatch.propTypes = {
  color: PropTypes.string.isRequired,
};

export default DocumentationSwatch;
