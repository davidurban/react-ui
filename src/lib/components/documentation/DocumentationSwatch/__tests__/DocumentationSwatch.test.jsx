import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import DocumentationSwatch from '../';

describe('rendering', () => {
  it('renders correctly mandatory props only', () => {
    const tree = shallow(<DocumentationSwatch color="primary" />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
