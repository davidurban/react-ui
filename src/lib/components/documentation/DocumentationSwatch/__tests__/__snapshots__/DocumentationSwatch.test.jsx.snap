// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly mandatory props only 1`] = `
<div
  className="root"
  title="primary"
>
  <div
    className="field"
    style={
      Object {
        "backgroundColor": "var(--rui-color-primary)",
      }
    }
  />
  <div
    className="label"
  >
    primary
  </div>
</div>
`;
